examples/SoftwareSerial/SoftwareSerial.ino

```diff
@@ -4,10 +4,18 @@
 #include <SoftwareSerial.h>
 #endif
 #include <PMserial.h> // Arduino library for PM sensors with serial interface
+
 #if !defined(PMS_RX) && !defined(PMS_TX)
-const uint8_t PMS_RX = 10, PMS_TX = 11;
+constexpr auto PMS_RX = 10;
+constexpr auto PMS_TX = 11;
 #endif
```

Comment:
Why is this better?
I think both of this statements would end up inlined on the pre-compilation or the compilation step.

Reply:
Just [see this](https://godbolt.org/z/7W8oaP):
Version with `"const"` generate **real** variable in **memory**.
Version with `define` generate normal optimized code.
(if you switch the language from C to C++ the situation will improve, but the standard does not guarantee this)

It's strange, but the "const" directive in C++ makes no guarantee that a constant will be used in the code. That's why we had to add constexpr in the new standard. For _guaranteed_ constants, constexpr must be used.

Reply:
Thanks for the reference.